README.md

```diff
@@ -776,6 +777,16 @@ These options should contain something like:
 - `PIWIK_URL=piwik.example.org`
 - `PIWIK_SITE_ID=42`
 
+#### Gitlab KAS
+
+Basic Gitlab KAS Settings 
+
+- `GITLAB_KAS_ENABLED=${GITLAB_KAS_ENABLED:-false}`
+- `GITLAB_KAS_SECRET=${GITLAB_INSTALL_DIR}/.gitlab_kas_secret}`
+- `GITLAB_KAS_EXTERNAL="wss://kas.example.com"}`
+- `GITLAB_KAS_INTERNAL="grpc://localhost:8153"}`
+- `GITLAB_KAS_PROXY=`
+
```

Comment:
I'm very sorry that I couldn't get enough time to explain it enough, but I hope you read this patch (and the result of applying it) to understand what you should have written in the README..

````patch
diff --git a/README.md b/README.md
index f8862af3..92e12bc7 100644
--- a/README.md
+++ b/README.md
@@ -777,16 +777,6 @@ These options should contain something like:
 - `PIWIK_URL=piwik.example.org`
 - `PIWIK_SITE_ID=42`
 
-#### Gitlab KAS
-
-Basic Gitlab KAS Settings 
-
-- `GITLAB_KAS_ENABLED=${GITLAB_KAS_ENABLED:-false}`
-- `GITLAB_KAS_SECRET=${GITLAB_INSTALL_DIR}/.gitlab_kas_secret}`
-- `GITLAB_KAS_EXTERNAL="wss://kas.example.com"}`
-- `GITLAB_KAS_INTERNAL="grpc://localhost:8153"}`
-- `GITLAB_KAS_PROXY=`
-
 #### Available Configuration Parameters
 
 *Please refer the docker run command options for the `--env-file` flag where you can specify all required environment variables in a single file. This will save you from writing a potentially long docker run command. Alternatively you can use docker-compose. docker-compose users and Docker Swarm mode users can also use the [secrets and config file options](#docker-secrets-and-configs)*
@@ -1093,6 +1083,26 @@ Default Google key file. Defaults to `$GITLAB_OBJECT_STORE_CONNECTION_GOOGLE_JSO
 
 Cron notation for the GitLab pipeline schedule worker. Defaults to `'19 * * * *'`
 
+##### `GITLAB_KAS_ENABLED`
+
+Enable/Disable GitLab agent server for Kubernetes (KAS). See details on [official documentation](https://docs.gitlab.com/ee/administration/clusters/kas.html). Defaults to `false`
+
+##### `GITLAB_KAS_SECRET`
+
+File that contains the secret key for verifying access for GitLab KAS. Defaults to `${GITLAB_INSTALL_DIR}/.gitlab_kas_secret`
+
+##### `GITLAB_KAS_EXTERNAL`
+
+User-facing URL for the in-cluster agent. Defaults to `"wss://kas.example.com"`
+
+##### `GITLAB_KAS_INTERNAL`
+
+Internal URL for the GitLab backend. Defaults to `"grpc://localhost:8153"`
+
+##### `GITLAB_KAS_PROXY`
+
+The URL to the Kubernetes API proxy (used by GitLab users). No default.
+
 ##### `GITLAB_LFS_ENABLED`
 
 Enable/Disable Git LFS support. Defaults to `true`.
````

Reply:
By the way, the official documentation describes that [you need to set an secret api key to `gitlab_kas['api_secret_key']` like:](https://docs.gitlab.com/ee/administration/clusters/kas.html#use-an-external-installation)

````ruby
gitlab_kas['api_secret_key'] = 'Your shared secret between GitLab and KAS'
````

Do you know anything about it? I have never used KAS and was interested in how it should be set from the user's point of view.

Reply:
I have made the suggested changes to the Readme,
I have no idea about the gitlab_kas['api_secret_key']

Reply:
Oh, sorry, I have sent you in a diff format. The "+" sign doesn't need to be included in the README.md. Just write like:

```md
##### `GITLAB_KAS_ENABLED`

Enable/Disable GitLab agent server for Kubernetes (KAS). See details on [official documentation](https://docs.gitlab.com/ee/administration/clusters/kas.html). Defaults to `false`

##### `GITLAB_KAS_SECRET`

File that contains the secret key for verifying access for GitLab KAS. Defaults to `${GITLAB_INSTALL_DIR}/.gitlab_kas_secret`

##### `GITLAB_KAS_EXTERNAL`

User-facing URL for the in-cluster agent. Defaults to `"wss://kas.example.com"`

##### `GITLAB_KAS_INTERNAL`

Internal URL for the GitLab backend. Defaults to `"grpc://localhost:8153"`

##### `GITLAB_KAS_PROXY`

The URL to the Kubernetes API proxy (used by GitLab users). No default.

```

Anyway, thank you for answering about the API key. I hope someone has some knowledge of KAS and can see if it works ...

Reply:
Updated to remove +

Reply:
Sorry for the many corrections, but as @sachilles says, this document looks inaccurate. I told you to remove it in the last review, but if you want to keep it, I think we should fix it and write a little more detail.

```suggestion
#### Gitlab KAS

GitLab agent server for Kubernetes (KAS) is disabled by default, but you can enable it by setting configuration parameter [`GITLAB_KAS_ENABLED`](#GITLAB_KAS_ENABLED) to true.
You can specify the internal/external URL for KAS via [`GITLAB_KAS_INTERNAL`](#GITLAB_KAS_INTERNAL) and [`GITLAB_KAS_EXTERNAL`](#GITLAB_KAS_EXTERNAL). If you set up proxy URL, use `GITLAB_KAS_PROXY`.

See official documentation : https://docs.gitlab.com/ee/administration/clusters/kas.html
```